### ISSUE TITLE
cannot override surface forcings using model diagnostics files as input 

### ISSUE THREAD

**Issue by nikizadehgfdl** [NOAA-GFDL/MOM6#322]
Fanrong tried to override a heat flux in MOM_surface_forcing.F90 by using a history file from a previous run of the same model (hence same grid), but the model came down with:

```
 FATAL from PE   192: data_override: data is not on data domain and compute domain is not divisible by size(data)

forrtl: error (76): Abort trap signal
Image              PC                Routine            Line        Source
fms_LOAR2E_exec.x  0000000002ED7306  mpp_mod_mp_mpp_er          50  mpp_util_mpi.inc
fms_LOAR2E_exec.x  00000000029C99C9  data_override_mod         674  data_override.F90
fms_LOAR2E_exec.x  00000000029CA1B5  data_override_mod         521  data_override.F90
fms_LOAR2E_exec.x  0000000000A57159  mom_surface_forci         721  MOM_surface_forcing.F90
fms_LOAR2E_exec.x  0000000000A516D8  mom_surface_forci         688  MOM_surface_forcing.F90
fms_LOAR2E_exec.x  000000000093BEB3  ocean_model_mod_m         385  ocean_model_MOM.F90
fms_LOAR2E_exec.x  0000000000401CE8  MAIN__                    916  coupler_main.F90
```

The data_table has one line:

```
"OCN", "hflx_adj"        , "hflx_adj"         ,"./INPUT/ocean_sfc_flx_adj.nc"       , "none" ,  1.0
```

The input file was made from a previous run diagnostics by

```
ncrename -v XH,xh -v YH,yh -v TIME,time -d TIME,time -d XH,xh -d YH,yh ocean_monthly.000101-010012.hflx_adj.nc\n",
ncrename -v HFLX_ADJ,hflx_adj -O /archive/td/LOAR2/LOAR2_Control_1860_E07/pp/ocean_monthly/ts/monthly/100yr/ocean_monthly.000101-010012.hflx_adj.nc\n",
```

srcDir is at (Matt's branch):

/lustre/f1/unswept/Fanrong.Zeng/LOAR2/libs_compile/src/mom6-examples/src/mom6/config_src/coupled_driver/MOM_surface_forcing.F90

WorkDir is at:

/lustre/f1/Fanrong.Zeng/work/LOAR2_Control_1990_E01_flxadjtest.o189274

**Comment by nikizadehgfdl** [comment]
Bill Hurlin says he had run into a similar issue with tau fluxes before and the following code fix suggested by Zhi bypassed the issue:

```
< call data_override('OCN', 'taux_adj', tempx_at_h, Time, override=overrode_x, &
>  call data_override('OCN', 'taux_adj', tempx_at_h(G%isc:G%iec,G%jsc:G%jec), Time, override=overrode_x, &
```

I have asked Fanrong to try a similar fix for hflx_adj to see if it works.

**Comment by adcroft** [comment]
Why was this closed? Was it fixed or not actually an issue?

**Comment by MJHarrison-GFDL** [comment]
This was fixed by 15439af9   

Should have included in the commit.
